vihreat-data/generate_ld/program.py

```diff
@@ -40,12 +40,50 @@ def _build_main(
 def _build_elements(markdown_path: str, parent_name: str) -> list[dict]:
     elements = []
     with open(markdown_path, "r", encoding="utf-8") as f:
+        previous_element_type = None
+        this_elements = None
+        n_inner_elements = 0
+        last_element = None
+        append = True
         for line in f:
             line = line.strip()
             if line:
-                elements.append(_build_element(line, parent_name, len(elements)))
+                this_element = _build_element(line, parent_name, len(elements) + n_inner_elements)
+                this_element_type = this_element[url.atomic("properties/isA")][0]
+                # Alkaako tässä kohtaa lista?
+                if previous_element_type != url.local("o/ActionItem") and this_element_type == url.local("o/ActionItem"):
+                    n_inner_elements += 1
+                    this_elements = [this_element]
+                    append = False
+                # Lista jatkuu
+                elif previous_element_type == url.local("o/ActionItem") and this_element_type == url.local("o/ActionItem"):
+                    n_inner_elements += 1
+                    this_elements += [this_element]
+                # Lista päättyy
+                elif previous_element_type == url.local("o/ActionItem") and this_element_type != url.local("o/ActionItem"):
```

Comment:
Jos ohjelma päättyy listaan, tuo lista jää nyt lisäämättä. (Esimerkiksi tietopoliittisessa ohjelmassa käy näin.)

Tämän voi tarkistaa loopin jälkeen, mutta yksinkertaisempaa on ehkä tehdä yksinkertaisesti tyyliin:

```
for line in f:
  e = build(line)
  if is_action_item(e):
    if is_action_list(elems[-1]):
      elems[-1][elements] += [e]
    else:
      elems.append(build_list(e))
  else:
    elems.append(e)
```